gnomad/utils/filtering.py

```diff
@@ -179,19 +181,36 @@ def filter_low_conf_regions(
             )
 
         telomeres_and_centromeres = resources.telomeres_and_centromeres.ht()
-        criteria.append(hl.is_missing(telomeres_and_centromeres[mt.locus]))
+        criteria.append(hl.is_missing(telomeres_and_centromeres[locus_expr]))
 
     if high_conf_regions is not None:
         for region in high_conf_regions:
             region = hl.import_locus_intervals(region)
-            criteria.append(hl.is_defined(region[mt.locus]))
+            criteria.append(hl.is_defined(region[locus_expr]))
 
     if criteria:
         filter_criteria = functools.reduce(operator.iand, criteria)
-        if isinstance(mt, hl.MatrixTable):
-            mt = mt.filter_rows(filter_criteria)
-        else:
-            mt = mt.filter(filter_criteria)
+        return filter_criteria
+    else:
+        raise ValueError("No low confidence regions requested for filtering.")
+
+
+def filter_low_conf_regions(
+    mt: Union[hl.MatrixTable, hl.Table],
```

Comment:
```suggestion
    t: Union[hl.MatrixTable, hl.Table],
```

this type gets called 't' elsewhere - having an 'mt' possible actually be an 'ht' is a bit odd!